### PR TITLE
Feat/esbuild

### PR DESCRIPTION
**Pull request by Tim-Tech-Dev**
### Replace `webpack5` with `esbuild`

- Improved compilation time
- `monaco-editor` v0.31.1
- esnext compatible
- Update esbuild config to include a watch and rebuild option
- Improved build script with easy to understand log outputs
- New update process: You only need to specify your bundle once.

### CODE REVIEW

**Comment by Tim-Tech-Dev** [comment]
Will need to add a watch mode and maybe production build.

File: nodecg-io-core/dashboard/esbuild.config.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+const prod = process.env.NODE_ENV === "production";
+
+const entryPoints = [
+    "monaco-editor/esm/vs/language/json/json.worker.js",
+    "monaco-editor/esm/vs/language/css/css.worker.js",
+    "monaco-editor/esm/vs/language/html/html.worker.js",
+    "monaco-editor/esm/vs/language/typescript/ts.worker.js",
+    "monaco-editor/esm/vs/editor/editor.worker.js",
+    "main.ts",
+];
```
Comment (daniel0611) [inline nodecg-io-core/dashboard/esbuild.config.js:21]: We only need the main editor and the json module of monaco. The rest can be safely removed as we're not using monaco to edit css, html or javascript.
This reduces the build output from 48,6 MiB to 21.0 MiB or from 8.2 MiB to 3.6 MiB if gzipped (as it is in tarballs/the npm registry).
```suggestion
const entryPoints = [
    "monaco-editor/esm/vs/language/json/json.worker.js",
    "monaco-editor/esm/vs/editor/editor.worker.js",
    "main.ts",
];
```

File: services/nodecg-io-debug/esbuild.config.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+const prod = process.env.NODE_ENV === "production";
+
+const entryPoints = [
+    "monaco-editor/esm/vs/language/json/json.worker.js",
+    "monaco-editor/esm/vs/language/css/css.worker.js",
+    "monaco-editor/esm/vs/language/html/html.worker.js",
+    "monaco-editor/esm/vs/language/typescript/ts.worker.js",
+    "monaco-editor/esm/vs/editor/editor.worker.js",
+    "dashboard/debug-helper.ts",
+];
```
Comment (daniel0611) [inline services/nodecg-io-debug/esbuild.config.js:21]: Same here
```suggestion
const entryPoints = [
    "monaco-editor/esm/vs/language/json/json.worker.js",
    "monaco-editor/esm/vs/editor/editor.worker.js",
    "dashboard/debug-helper.ts",
];
```

**Review by daniel0611 — APPROVED** [review]
Thank you so much for this

**Pull request merged**